backend/market/views.py

```diff
@@ -57,167 +68,163 @@ def get_queryset(self):
-        if starts_before:
-            queryset = queryset.filter(start_date__lt=starts_before)
-        if starts_after:
-            queryset = queryset.filter(start_date__gt=starts_after)
-        if ends_before:
-            queryset = queryset.filter(end_date__lt=ends_before)
-        if ends_after:
-            queryset = queryset.filter(end_date__gt=ends_after)
-        if min_price:
-            queryset = queryset.filter(price__gte=min_price)
-        if max_price:
-            queryset = queryset.filter(price__lte=max_price)
-        if negotiable:
-            queryset = queryset.filter(negotiable=negotiable)
-        if beds:
-            queryset = queryset.filter(beds=beds)
-        if baths:
-            queryset = queryset.filter(baths=baths)
-
-        record_analytics(Metric.SUBLET_BROWSE, request.user.username)
+        # Define filter mappings
+        filter_mappings = {
+            "title": "item__title__icontains",
+            "min_price": "item__price__gte",
+            "max_price": "item__price__lte",
+            "negotiable": "item__negotiable",
+            "address": "address__icontains",
+            "beds": "beds",
+            "baths": "baths",
+            "start_date_min": "start_date__gte",
+            "start_date_max": "start_date__lte",
+            "end_date_min": "end_date__gte",
+            "end_date_max": "end_date__lte",
+        }
+
+        # Apply filters using the helper function
+        queryset = apply_filters(
+            queryset=queryset,
+            params=params,
+            filter_mappings=filter_mappings,
+            user=request.user,
+            is_sublet=True,
+            tag_field="item__tags__name",
+        )
 
         # Serialize and return the queryset
-        serializer = SubletSerializerSimple(queryset, many=True)
+        serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
 
 class CreateImages(generics.CreateAPIView):
-    serializer_class = SubletImageSerializer
+    serializer_class = ItemImageSerializer
     http_method_names = ["post"]
-    permission_classes = [SubletImageOwnerPermission | IsSuperUser]
+    permission_classes = [ItemImageOwnerPermission | IsSuperUser]
     parser_classes = (
         MultiPartParser,
         FormParser,
     )
 
     def get_queryset(self, *args, **kwargs):
-        sublet = get_object_or_404(Sublet, id=int(self.kwargs["sublet_id"]))
-        return SubletImage.objects.filter(sublet=sublet)
+        item = get_object_or_404(Item, id=int(self.kwargs["item_id"]))
+        return ItemImage.objects.filter(item=item)
 
     # takes an image multipart form data and creates a new image object
     def post(self, request, *args, **kwargs):
         images = request.data.getlist("images")
-        sublet_id = int(self.kwargs["sublet_id"])
-        self.get_queryset()  # check if sublet exists
+        item_id = int(self.kwargs["item_id"])
+        self.get_queryset()  # check if item exists
```

Comment:
uh idk what this self.get_queryset() is doing?

Reply:
It calls the custom get_queryset defined above. If there is no Item with the inputted ID, self.get_queryset will return a `404 Not Found`.